include/superior_mysqlpp/extras/prepared_statement_utils.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <superior_mysqlpp/connection.hpp>
+#include <superior_mysqlpp/prepared_statement.hpp>
+#include <superior_mysqlpp/utils.hpp>
+#include <type_traits>
+
+namespace SuperiorMySqlpp
+{
+    namespace detail
+    {
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...) const volatile) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...) volatile) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...) const) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        template <typename CallableClass, typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (CallableClass::*)(Args...)) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+
+        template <typename CallableClass>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, CallableClass&&) {
+            // This is actually only callable argument type deductor. We don't care about callable type, so we can decay Callable object
+            return generatePreparedStatementImpl(connection, query, &std::decay_t<CallableClass>::operator());
+        }
+
+        template <typename... Args>
+        auto generatePreparedStatementImpl(Connection &connection, const std::string &query, void (*)(Args...)) {
+            return connection.makePreparedStatement<ResultBindings<std::remove_cv_t<std::remove_reference_t<Args>>...>>(query);
+        }
+    }
+
+    /**
+     * Executes query, reads input data and passes them to callback function
+     */
+    template<typename PreparedStatementType, typename Callable>
+    void psReadQuery(PreparedStatementType &&ps, Callable &&processingFunction)
+    {
+        try
+        {
+            ps.execute();
+            while(ps.fetch())
+            {
+                invokeViaTuple(processingFunction, ps.getResult());
+            }
+        } catch(...)
+        {
+            std::cout << "ERROR while processing query" << std::endl;
```

Comment:
I'm not sure, this is correct error reporting. Isn't there standard log facility?